service/frontend/workflowHandler.go

```diff
@@ -491,6 +495,19 @@ func (wh *WorkflowHandler) PollForDecisionTask(
 	return resp, nil
 }
 
+func (wh *WorkflowHandler) checkBadBinary(domainEntry *cache.DomainCacheEntry, binaryChecksum string) error {
+	if domainEntry.GetConfig().BadBinaries.Binaries != nil {
+		badBinaries := domainEntry.GetConfig().BadBinaries.Binaries
+		_, ok := badBinaries[binaryChecksum]
+		if ok {
+			return &gen.BadRequestError{
```

Comment:
I think we should fail the decision and also return error back to user for logging.

Reply:
Done